### PR TITLE
Clean Intersect_2 in Triangulation_2_projection_traits_3

### PR DESCRIPTION
**Pull request by sloriot**
Clean the `Intersect_2` functor and add the case of overlap between segments

### CODE REVIEW

**Comment by lrineau** [comment]
For 5.2-beta2 or 5.3?

File: Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h [hunk 1]
```diff
@@ -202,36 +202,94 @@ class Projected_intersect_3
...
-        // the cross product of s1.to_vector() and s2.to_vector(). That
-        // plane should intersect *l, now.
-        return intersection(*line, Plane_3(s1.source(),
-                                           cross_product(s1.to_vector(),
-                                                         s2.to_vector())));
+        // Let the plane passing through s1.source() and with normal the traits' normal.
+        // The segment intersection point computed as the intersection
+        // of the *l and that plane. Note that if segments are not coplanar the positive
+        // of the intersection point is somehow arbitrary.
+        return intersection(*line, Plane_3(s1.source(), normal));
```
Comment (sloriot) [inline Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h:234]: I think this change is buggy

**Comment by maxGimeno** [comment]
errors (and conflicts) : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-58/Linear_cell_complex_Demo/TestReport_lrineau_ArchLinux-CXX17-Release.gz

File: Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h [hunk 2]
```diff
@@ -202,21 +202,28 @@ class Projected_intersect_3
...
-         cross_product(normal, pi - s2.source())
+           cross_product(normal, pi - s2.source())
          * cross_product(normal, pi - s2.target()) > FT(0) )
       {
-        // the intersection of the lines is not inside the segments
+        // the intersection of the supporting lines is not inside both segments
+#ifdef CGAL_T2_PTB_3_DEBUG
         std::cerr << "intersection not inside\n";
         return Object();
+#endif
```
Comment (MaelRL) [inline Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h:226]: The debug macro including the `return` looks :bug:-y.

File: Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h [hunk 3]
```diff
@@ -203,21 +204,29 @@ class Projected_intersect_3
...
+           cross_product(normal, pi - s2.source())
          * cross_product(normal, pi - s2.target()) > FT(0) )
       {
-        // the intersection of the lines is not inside the segments
+        // the intersection of the supporting lines is not inside both segments
+#ifdef CGAL_T2_PTB_3_DEBUG
         std::cerr << "intersection not inside\n";
         return boost::none;
+#endif
+        return boost::none;
```
Comment (lrineau) [inline Triangulation_2/include/CGAL/Triangulation_2/internal/Triangulation_2_projection_traits_base_3.h:229]:
```suggestion
#ifdef CGAL_T2_PTB_3_DEBUG
        std::cerr << "intersection not inside\n";
#endif
        return boost::none;
```

**Comment by maxGimeno** [comment]
tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-15.shtml

**Pull request merged**